be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -99,17 +99,29 @@ Status HdfsScannerCSVReader::_fill_buffer() {
 Status HdfsTextScanner::do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     TTextFileDesc text_file_desc = _scanner_params.scan_ranges[0]->text_file_desc;
     _field_delimiter = text_file_desc.field_delim;
-    // we should cast string to char now since csv reader only support record delimiter by char
+
+    // we should cast string to char now since csv reader only support record, collection, mapkey delimiter by char
     _record_delimiter = text_file_desc.line_delim.front();
+    _collection_delimiter = text_file_desc.collection_delim.front();
```

Comment:
Is delimiter only allowed to be one char?

Reply:
Yes, I have already checked it.
You can refer to: https://github.com/apache/hive/blob/master/serde/src/java/org/apache/hadoop/hive/serde2/lazy/LazySerDeParameters.java#L251
The separator only occupied one byte.